### PR TITLE
Add CIS GCP 2.1

### PR DESCRIPTION
**Pull request by orouz**
### Summary of your changes

Adds CIS GCP 2.1

##### missing DATA_READ and DATA_WRITE (both need to be inherited or set on a project/folder/org)
| failed  |  passed   |
|-----------|---|
| Screenshot 2023-09-21 at 17 46 39 | Screenshot 2023-09-21 at 17 45 06 |

### Related Issues
 - closes https://github.com/elastic/csp-security-policies/issues/305

### Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?
- [x] Generate rule metadata using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rules-metadata)
- [x] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

> **Note** Make sure to bump cloudbeat [`version/policy.go`](https://github.com/elastic/cloudbeat/blob/main/version/policy.go#L20) after merging this PR and drafting a new release.

### CODE REVIEW

File: bundle/compliance/cis_gcp/rules/cis_2_1/test.rego [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import data.cis_gcp.test_data
+import data.compliance.policy.gcp.data_adapter
+import data.lib.test
+
+type := "project-management"
+
+subtype := "gcp-cloudresourcemanager-project"
+
+test_violation {
```
Comment (orouz) [inline bundle/compliance/cis_gcp/rules/cis_2_1/test.rego:11]: TODO: add a test case for multiple `auditLogConfigs ` entries where 1 member has both read/write and the other doesnt

**Comment by orouz** [comment]
putting this on hold until we support `org` level scoping

File: bundle/compliance/cis_gcp/rules/cis_2_1/rule.rego [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	has_read_write_logs(policy)
+	not has_exempted_members(policy)
+} else = false
+
+has_read_write_logs(policy) if {
+	log_types := {t | t = policy.audit_configs[i].audit_log_configs[j].log_type}
+	2 in log_types # "DATA_WRITE"
+	3 in log_types # "DATA_READ" 
+	policy.audit_configs[_].service == "allServices"
+} else = false
```
Comment (orouz) [inline bundle/compliance/cis_gcp/rules/cis_2_1/rule.rego:30]: at least 1 policy with both read/write log types configured for all services is required in order to pass. can be either org/folder/project policy. 

a policy that inherits a single log type and has the other one configured on its own will fail.

File: bundle/compliance/cis_gcp/rules/cis_2_1/rule.rego [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		common.calculate_result(cloud_logging_is_configured),
+		input.resource,
+	)
+}
+
+cloud_logging_is_configured if {
+	policy := input.resource[_].iam_policy
+	has_read_write_logs(policy)
+	not has_exempted_members(policy)
+} else = false
```
Comment (oren-zohar) [inline bundle/compliance/cis_gcp/rules/cis_2_1/rule.rego:23]: you already set it as `false` in line 8

File: bundle/compliance/cis_gcp/rules/cis_2_1/rule.rego [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+has_read_write_logs(policy) if {
+	log_types := {t | t = policy.audit_configs[i].audit_log_configs[j].log_type}
+	2 in log_types # "DATA_WRITE"
+	3 in log_types # "DATA_READ"
```
Comment (oren-zohar) [inline bundle/compliance/cis_gcp/rules/cis_2_1/rule.rego:28]: what about `ADMIN_READ`? 

> Ensure that **Admin Read**, Data Write, and Data Read are enabled for all Google Cloud services and that no exemptions are allowed.

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: bundle/compliance/cis_gcp/rules/cis_2_1/rule.rego (hunk 4 above)
Comment (orouz) [reply bundle/compliance/cis_gcp/rules/cis_2_1/rule.rego:28]: thanks. will add. missed it. a bit odd it's part of the manual audit but not the automated one.

**Review by oren-zohar — APPROVED** [review]

**Comment by oren-zohar** [comment]
@orouz please don't forget to add all the missing assets to our integration

**Pull request merged**